specs/phase0/fork-choice.md

```diff
@@ -459,3 +464,23 @@ def on_attestation(store: Store, attestation: Attestation, is_from_block: bool=F
     # Update latest messages for attesting indices
     update_latest_messages(store, indexed_attestation.attesting_indices, attestation)
 ```
+
+#### `on_attester_slashing`
+
+```python
+def on_attester_slashing(store: Store, attester_slashing: AttesterSlashing) -> None:
+    """
+    Run ``on_attester_slashing`` upon receiving a new ``AttesterSlashing`` from either within a block or directly 
+    on the wire.
+    """
+    attestation_1 = attester_slashing.attestation_1
+    attestation_2 = attester_slashing.attestation_2
+    assert is_slashable_attestation_data(attestation_1.data, attestation_2.data)
+    state = store.block_states[store.justified_checkpoint.root]
+    assert is_valid_indexed_attestation(state, attestation_1)
+    assert is_valid_indexed_attestation(state, attestation_2)
+
+    indices = set(attestation_1.attesting_indices).intersection(attestation_2.attesting_indices)
+    for index in sorted(indices):
```

Comment:
why sorted? `has_equivocated` is a set so it doesn't really matter the order they are inserted in